diff -Nur dc_gui2-0.68.orig/src/callbacks.c dc_gui2-0.68.chng/src/callbacks.c
--- dc_gui2-0.68.orig/src/callbacks.c	Sat Jan  4 08:24:38 2003
+++ dc_gui2-0.68.chng/src/callbacks.c	Wed Jan  8 13:54:11 2003
@@ -560,7 +560,11 @@
 
 	s=g_string_new(NULL);
 	path=getenv("HOME");
-	g_string_sprintf(s,"%s/.dctc/recent",(path!=NULL)?path:".");
+	if (getenv("CONFIG_DIR"))
+		g_string_sprintf(s, "%s/%s/dctc/recent",
+		    (path!=NULL)?path:".", getenv("CONFIG_DIR"));
+	else
+		g_string_sprintf(s, "%s/.dctc/recent", (path!=NULL)?path:".");
 	unlink(s->str);
 	g_string_free(s,TRUE);
 	fill_recent_hub_clist();
diff -Nur dc_gui2-0.71.orig/src/main.c dc_gui2-0.71.chng/src/main.c
--- dc_gui2-0.71.orig/src/main.c	Sun Mar  9 09:12:31 2003
+++ dc_gui2-0.71.chng/src/main.c	Wed Mar 19 09:21:11 2003
@@ -414,7 +414,12 @@
 
 	/* main dir= $HOME/.dctc */
 	dctc_main_dir=g_string_new(NULL);
-	g_string_sprintf(dctc_main_dir,"%s/.dctc",(path!=NULL)?path:".");
+	if (getenv("CONFIG_DIR"))
+		g_string_sprintf(dctc_main_dir, "%s/%s/dctc",
+		    (path!=NULL)?path:".", getenv("CONFIG_DIR"));
+	else
+		g_string_sprintf(dctc_main_dir, "%s/.dctc", (path!=NULL)?path:".");
+	
 	if(access(dctc_main_dir->str,R_OK|W_OK|X_OK))
 	{
 		if(errno==ENOENT)
diff -Nur dc_gui2-0.68.orig/src/recent_hub_clist.c dc_gui2-0.68.chng/src/recent_hub_clist.c
--- dc_gui2-0.68.orig/src/recent_hub_clist.c	Wed Jan  1 15:24:57 2003
+++ dc_gui2-0.68.chng/src/recent_hub_clist.c	Wed Jan  8 13:57:01 2003
@@ -46,7 +46,11 @@
 
 	s=g_string_new(NULL);
 	path=getenv("HOME");
-	g_string_sprintf(s,"%s/.dctc/recent",(path!=NULL)?path:".");
+	if(getenv("CONFIG_DIR"))
+		g_string_sprintf(s, "%s/%s/dctc/recent",
+		    (path!=NULL)?path:".", getenv("CONFIG_DIR"));
+	else
+		g_string_sprintf(s, "%s/.dctc/recent",(path!=NULL)?path:".");
 
 	/* we read the file ~/.dctc/recent */
 	f=fopen(s->str,"rb");
